Orange/data/util.py

```diff
@@ -28,3 +28,27 @@ def scale(values, min=0, max=1):
     if ptp == 0:
         return np.clip(values, min, max)
     return (-minval + values) / ptp * (max - min) + min
+
+
+class ComputeValueWCommon:
+    """This class stuctures compute_value functions that
+    repeat parts of computation for different variables into
+    common and specific parts.
+
+    Attributes:
+        commonfn    Function that performs computation that is common
+                    for multiple variables. All variables that share
+                    computation need to set the same function instance.
+    """
+    commonfn = lambda data: None
+
+    def __call__(self, data, common=None):
+        """Fallback if common parts are not passed."""
+        if common is None:
+            common = self.commonfn(data)
+        return self.specific(common, data)
+
+    def specific(self, common, data):
+        """Given the common computation results, perform variable-specific
+        part of computation and return new variable values."""
+        raise NotImplementedError("specific computation must be implemented")
```

Comment:
I agree with @kernc. In general, I don't like the style in which we directly address the user or developer. Many errors/warnings in widgets used to say something like "Too few features. You need to provide at least two features to thingamagog". I prefer "Thingamagog needs at least two features."

In this case, even if you keep the message, rephrase it to "Method 'specific' is not implemented" (but this is obvious) or "... must be implemented", or something along this line. Just without "you". :)